releng/handbook: Add documentation for image promotion during the release process (#1166)
kubernetes/sig-release: release-engineering/role-handbooks/branch-manager.md

### Beta Releases

> Before run the `nomock` please refeer to the [Image Promotion documentation](#image-promotion).

Reviewer: ```suggestion
Before run the `nomock stage` please refer to the [Image Promotion documentation](#image-promotion).
```

**Note: If this is the first release (`rc.0`), there are additional tasks to complete. Please review them _COMPLETELY_ in the [Branch Creation section](#branch-creation), _before_ continuing.**

> Before run the `nomock` please refeer to the [Image Promotion documentation](#image-promotion).

Reviewer: ```suggestion
Before run the `nomock stage` please refer to the [Image Promotion documentation](#image-promotion).
```

### Official Releases

> Before run the `nomock` please refeer to the [Image Promotion documentation](#image-promotion).

Reviewer: ```suggestion
Before run the `nomock stage` please refer to the [Image Promotion documentation](#image-promotion).
```

3. CHANGELOG-X.Y.md is automatically loaded into the kubernetes/kubernetes repo: [https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.16.md](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.16.md)

### Image Promotion

>  We need publishing images to the `k8s-staging-kubernetes` and then promoting them to prodution. To do that we need to follow the steps:

Reviewer: ```suggestion
 We need publishing images to the `k8s-staging-kubernetes` GCS Bucket and then promoting them to production. To do that we need to follow the steps:
```

 This step should be done after `Mock Stage` and `Mock Release` are completed and no errors.

 - Forking/cloning https://github.com/kubernetes-sigs/k8s-container-image-promoter
 - Forking/cloning https://github.com/kubernetes/k8s.io. If you already have that make sure you have an up-to-date copy of the repository.
 - From the root of `k8s-container-image-promoter` repository

 ```shell

>  go install ./cmd/cip-mm/... # Installs the manifest merge utility, cip-mm

Reviewer: Can we `get get …` the promoter? If so, then I would prefer this way.

Author: tried but got
```
GO111MODULE=on go get sigs.k8s.io/k8s-container-image-promoter/cmd/cip-mm
go get sigs.k8s.io/k8s-container-image-promoter/cmd/cip-mm: module sigs.k8s.io/k8s-container-image-promoter@upgrade found (v1.0.7), but does not contain package sigs.k8s.io/k8s-container-image-promoter/cmd/cip-mm
```

 ```

 - From the root of `k8s.io` repository

 ```shel
 cip-mm --base_dir=$GOPATH/src/k8s.io/k8s.io/k8s.gcr.io --staging_repo=gcr.io/k8s-staging-kubernetes --filter_tag=v1.19.0-rc.2
 ```

 _note_: If you clone the repo in another structure like `$GOPATH/src/github.com/kubernetes/k8s.io` you will nee to set the `--base_dir` as `--base_dir=$GOPATH/src/github.com/kubernetes/k8s.io/k8s.gcr.io`
_note 2_: The `--filter_tag` you will add the release that you are working, like `v1.19.0-rc.3` / `v1.19.0-beta.1` / `v1.19.0`

 - After the command ran you should see updates to the `k8s.gcr.io/images/k8s-staging-kubernetes/images.yaml`
 - Create a branch and open a Pull Request with the new digests to promote. Add the links for cloudbuild jobs for `Mock Stage` and `Mock Release` and tag `@kubernetes/release-engineering` and add a explict `/hold` to be lifted before we start the `Official Release` step.

 Examples:
 - [PR example 1](https://github.com/kubernetes/k8s.io/pull/1049)
 - [PR example 2](https://github.com/kubernetes/k8s.io/pull/1071)

>  - After the Pull Request is approved and **before** start the `Official Release` step, we need to unhold the PR to get that merged  and watch the following [Prow Job](https://prow.k8s.io/?job=post-k8sio-cip) when the latest master ran without errors we can kick off the `Official Release`.

Reviewer: ```suggestion
 - After the Pull Request is approved and **before** start the `Official Release` step, we need to unhold the PR to get that merged  and watch the following [Prow Job](https://prow.k8s.io/?job=post-k8sio-cip) to succeed. When the latest master ran without errors, then we can continue with `Official Release`.
```
